src/protocol/prf_sharding/mod.rs

```diff
@@ -677,27 +679,33 @@ where
     F: PrimeField + ExtendableField,
 {
     let mut step: usize = 1 << BK::BITS;
-    let mut row_contribution = vec![value; 1 << BK::BITS];
+    let total_breakdowns = match breakdown_count {
+        Some(x) => x,
+        None => 1 << BK::BITS,
+    };
+    assert!(total_breakdowns <= 1 << BK::BITS);
+    let mut row_contribution = vec![value; total_breakdowns];
 
     for (tree_depth, bit_of_bdkey) in bd_key.iter().rev().enumerate() {
         let depth_c = ctx.narrow(&BinaryTreeDepthStep::from(tree_depth));
         let span = step >> 1;
-        let mut futures = Vec::with_capacity((1 << BK::BITS) / step);
-        for i in (0..1 << BK::BITS).step_by(step) {
+        let mut futures = Vec::with_capacity(bd_key.len() / step);
```

Comment:
This doesn't seem right. 
Consider the loop when `step = 4`;
`bd_key.len() / step` is something like `5 / 4`, which is ~= 1. Previously we initialized the number of futures to something like `2^5 / 4`, which is 8.

It works because the vector resizes, but the idea is to avoid that.

Reply:
my bad, this got added unnecessarily, fixing it